source/kraken/apply_disruption.cpp

```diff
@@ -201,6 +195,149 @@ struct add_impacts_visitor : public apply_impacts_visitor {
+
+            // Computing a validity_pattern of impact used to pre-filter concerned vjs later
+            // Loop on every (even partial) days of the period
+            type::ValidityPattern impact_vp{meta.production_date.begin()}; // bitset are all initialised to 0
+            for (const auto& period: impact->application_periods) {
+                for (bg::day_iterator it(period.begin().date()); it <= period.end().date() ; ++it) {
+                    if (! meta.production_date.contains(*it)) { continue; }
+                    auto day = (*it - meta.production_date.begin()).days();
+                    impact_vp.add(day);
+                }
+            }
+
+            // Loop on impacted routes of the line section
+            for(auto* route: ls.routes) {
+                // Loop on each vj
+                bool has_affected_vj(false);
+                route->for_each_vehicle_journey([&](nt::VehicleJourney& vj) {
+                    /*
+                     * Pre-filtering by validity pattern, which allows us to check if the vj is impacted quickly
+                     *
+                     * Since the validity pattern runs only by day not by hour, we'll compute in detail to
+                     * check if the vj is really impacted or not.
+                     *
+                     * */
+                    if ((vj.validity_patterns[rt_level]->days & impact_vp.days).none()) {
+                        return true;
+                    }
+
+                    // Filtering each journey to see if it's impacted by the section.
+                    SectionBounds bounds_st = vj.get_bounds_stop_times_for_section(ls.start_point, ls.end_point);
+                    // If the vj pass by both stops both elements will be different than nullptr, otherwise
+                    // it's not passing by both stops and should not be impacted
+                    if(bounds_st.first && bounds_st.second) {
+                        // Once we know the line section is part of the vj we compute the vp for the adapted_vj
+                        LOG4CPLUS_TRACE(log, "vj "<< vj.uri << " pass by both stops, might be affected.");
+                        nt::ValidityPattern new_vp{vj.validity_patterns[rt_level]->beginning_date};
+                        for(const auto& period : impact->application_periods) {
+                            // get the vp of the section
+                            new_vp.days |= vj.get_vp_for_section(bounds_st, rt_level, period).days;
+                        }
+                        // If there is effective days for the adapted vp we're keeping it
+                        if(!new_vp.days.none()){
+                            LOG4CPLUS_TRACE(log, "vj "<< vj.uri << " is affected, keeping it.");
+                            new_vp.days >>= vj.shift;
+                            has_affected_vj = true;
+                            vj_vp_pairs.emplace_back(&vj, new_vp, bounds_st);
```

Comment:
Hum maybe i'm missing something but this seems quite error prone.

we take an address from a vehicle_journey in the route's vj's vector.

but afterwards, in the impact application (line 311), we create a new vj with the modification and we add the new vj in the route's vector.

We thus can activate a vector reallocation and invalidate the other vj's addresses :confused: no ?